cpp/src/slice2cs/CsUtil.cpp

```diff
@@ -888,6 +888,31 @@ Slice::CsGenerator::writeUnmarshalCode(Output &out,
     }
 }
 
+namespace
+{
+bool isVariableLength(TypePtr type)
+{
+    BuiltinPtr builtin = BuiltinPtr::dynamicCast(type);
+    ProxyPtr proxy = ProxyPtr::dynamicCast(type);
+
+    if(builtin &&
+       builtin->kind() != Builtin::KindValue &&
+       builtin->kind() != Builtin::KindObjectProxy &&
+       builtin->kind() != Builtin::KindString)
+    {
+        return false;
+    }
+    StructPtr st = StructPtr::dynamicCast(type);
+    if(st && !st->isVariableLength())
+    {
+        return false;
+    }
+
+    return true;
+}
```

Comment:
It's not clear from the method name what this is useful for. It could use a better name or comment. I wonder if it's also correct. We don't use the proxy variable and why don't we just use `type->isVariableLength()`? Why does it need to return something different? If we keep it we'll have to remember to add support for VarInt etc.

Reply:
Maybe I can just use `type->isVariableLength` I wasn't aware it was implemented all types.

Reply:
Fixed